SandWorm/Components/BaseKinectComponent.cs

```diff
@@ -0,0 +1,196 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Grasshopper.Kernel;
+using Microsoft.Kinect;
+using Rhino;
+using Rhino.Geometry;
+
+namespace SandWorm.Components
+{
+    // Provides common functions across the components that read from the Kinect stream
+    public abstract class BaseKinectComponent : BaseComponent
+    {
+        // Common Input Parameters
+        public int averageFrames = 1;
+        public int blurRadius = 1;
+        public int keepFrames = 1;
+        // Sandworm Options
+        public List<double> options; // List of options coming from the SWSetup component
```

Comment:
While working on the MeshCompare logic I realized that we need to be able to pass data between components, which is a bit more differentiated than a list of double. [Here](https://github.com/mariuszhermansdorfer/SandWorm/blob/feature/cutfill_rebased/SandWorm/SetupOptions.cs) is my attempt at grouping these. 
Would you incorporate this later on?

Reply:
Happy to incorporate it now to try and minimise merge problems down the line. I've added to this branch as 04f083419e76d4e1a0ae9eb1781730484bc905bf.